### PR TITLE
Apply gcc specifc warnflags only to gcc

### PR DESCRIPTION
**Pull request by oflebbe**
The approach of pull request #95 didn't work for me. This pull  is less invasive.

### CODE REVIEW

File: configure.in [hunk 1]
```diff
@@ -65,6 +66,10 @@ AS_IF([test -z "$krb5config"],
 	[ krb5config="$PATH_KRB5_CONFIG"]
 ])
 
+if test x$GCC = xyes ; then
+  WARNFLAGS="-Wall -Wextra -Wno-write-strings"
+fi
```
Comment (dkobras) [inline configure.in]: Do we really want to run without any `WARNFLAGS` at all for non-GCC compilers? At least -Wall should be understood universally, shouldn't it?

Comment (oflebbe) [inline configure.in]: Nope. -Wall isn't understood by aCC AFAIK

Comment (oflebbe) [inline configure.in]: But why both exotic platforms with warnings in the first place?

**Comment by michael-o** [comment]
Why this is really less invasive, those options can apply to Solaris' or any other commercial Unix compiler. aCC doesn't. So the standard macros where the best approach.

I can live with this, but it doesn't feel complete and works on HP-UX for me.

**Comment by oflebbe** [comment]
@dkobras On MacOS it will detect clang as gcc, so it will use the warning flags as well.

**Pull request closed without merging**